tests/pubsub.rs

```diff
@@ -54,25 +55,30 @@ async fn publish_between_two_nodes() {
     use std::collections::HashSet;
 
     let nodes = spawn_nodes(2, Topology::Line).await;
+    let node_a = &nodes[0];
+    let node_b = &nodes[1];
 
-    let topic = "shared".to_owned();
+    let topic_a = "shared-a".to_owned();
+    let topic_b = "shared-b".to_owned();
 
-    let mut a_msgs = nodes[0].pubsub_subscribe(topic.clone()).await.unwrap();
-    let mut b_msgs = nodes[1].pubsub_subscribe(topic.clone()).await.unwrap();
+    // Node A subscribes to Topic B
+    // Node B subscribes to Topic A
+    let mut a_msgs = node_a.pubsub_subscribe(topic_b.clone()).await.unwrap();
+    let mut b_msgs = node_b.pubsub_subscribe(topic_a.clone()).await.unwrap();
```

Comment:
Unclear why this was done. Is it now impossible to have the two peers in the same topic and see each other in the peers...? I think this should be investigated. I'll check to see if this was made before the partial_view change, or after it.

Reply:
I mainly changed the test to confirm my hypothesis about which messages were actually being received. At the time, I didn't know which node had what PeerId. After altering the test, I discovered the only messages received were from the localhost.

I'm not opposed to reverting the commit. We would still want the timeout on the `.take`.

Reply:
I think the failed conformance tests could be related to this. Except that they only failed on windows which I don't have around for development. I'll look around, and rerun the windows build.

Reply:
Sadly no luck with the re-run, however the failing tests did change: https://github.com/rs-ipfs/rust-ipfs/runs/5754868818?check_suite_focus=true#step:18:6049 -- need to continue with this tomorrow.

Reply:
Been looking at this a bit. I think you had a good idea with the test case, so it should be kept next to the old one. I cannot really reproduce the failure on the old version, but did some simplifications (from hashset -> vec) and expect all messages (even the echoed locally sent ones). Cannot fathom why hashset was used there originally, but I recall it was added my be.